core/whistle_apps-1.0.0/src/whapps_alert.erl

```diff
@@ -0,0 +1,130 @@
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec enabled() -> boolean().
+enabled() ->
+    whapps_config:get_is_true(?CONFIG_CAT, <<"enabled">>, 'true').
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec fetch(ne_binary()) -> {'ok', wh_json:object()} |
+                            {'error', any()}.
+fetch(AlertId) ->
+    kzd_alert:fetch(AlertId).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec create(ne_binary(), ne_binary(), wh_json:object(), wh_json:object()) -> {'ok', wh_json:object()} |
+                                                                              {'error', any()}.
+-spec create(ne_binary(), ne_binary(), wh_json:object()
+             ,wh_json:object(), wh_proplist()) -> {'ok', wh_json:object()} |
+                                                  {'error', any()}.
+create(Title, Message, From, To) ->
+    create(Title, Message, From, To, []).
+
+create('undefined', _Message, _From, _To, _Opts) ->
+    {'required', kzd_alert:title()};
+create(_Title, 'undefined', _From, _To, _Opts) ->
+    {'required', kzd_alert:message()};
+create(_Title, _Message, 'undefined', _To, _Opts) ->
+    {'required', kzd_alert:from()};
+create(_Title, _Message, _From, 'undefined', _Opts) ->
+    {'required', kzd_alert:to()};
+create(Title, Message, From, To, Opts) ->
+    case enabled() of
+        'false' -> {'error', 'disabled'};
+        'true' ->
+            Routines = [
+                fun(J) -> kzd_alert:set_title(J, Title) end
+                ,fun(J) -> kzd_alert:set_message(J, Message) end
+                ,fun(J) -> kzd_alert:set_from(J, From) end
+                ,fun(J) -> kzd_alert:set_to(J, To) end
+                ,fun(J) -> maybe_add_options(J, Opts) end
+            ],
+            {'ok', lists:foldl(fun(F, J) -> F(J) end, kzd_alert:new(), Routines)}
+    end.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec save(wh_json:object()) -> {'ok', wh_json:object()} | {'error', any()}.
+save(JObj) ->
+    case enabled() of
+        'false' -> {'error', 'alerts_disabled'};
+        'true' ->
+            couch_mgr:save_doc(?WH_ALERTS_DB, JObj)
+    end.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec delete(wh_json:object()) -> {'ok', wh_json:object()} | {'error', any()}.
+delete(AlertId) when is_binary(AlertId) ->
+    case ?MODULE:fetch(AlertId) of
```

Comment:
Why `?MODULE`? Do you really intend for this code to use only the latest version of fcall?